corehq/apps/cloudcare/static/cloudcare/js/utils.js

```diff
@@ -371,13 +371,23 @@ hqDefine('cloudcare/js/utils', [
         return inputDate;
     };
 
-    var initDatePicker = function ($el, selectedDate, dateFormat) {
+    var dateFormat = 'MM/DD/YYYY';
+    var dateFormats = ['MM/DD/YYYY', 'YYYY-MM-DD', 'M/D/YYYY', 'M/D/YY', 'M-D-YYYY', 'M-D-YY', moment.defaultFormat];
+
+    /** Coerce an input date string to a moment object */
+    var parseInputDate = function (dateString) {
```

Comment:
I feel like something weird happens if the dateString input is `YYYY-MM-DD` since `convertTwoDigitYear ` will try to convert the DD into four digits. But the unit test seems to pass and i can't figure out why

Reply:
Ah, no issue here. `YYYY-MM-DD` won't be converted because `convertTwoDigitYear` checks if `dateString` has <= 6 digits before converting

Reply:
Nice, yeah that makes sense - it might even be reasonable to switch to a more strict pattern like regex matching `^\d{1,2}[.-/]\d{1,2}[.-/]\d{1,2}$`.  Hell, I wonder if temporarily patching `moment.parseTwoDigitYear` might be an option if we did something like:
```js
oldParseTwoDigitYear = moment.parseTwoDigitYear;
moment.parseTwoDigitYear = function () {...};
moment.parseTwoDigitYear = oldParseTwoDigitYear
```

Anyways, I think iterating via unit tests is probably our best bet here, especially given that part of this is controlled somewhat opaquely by a library.  Once I switched to that approach, I felt a lot more confident in this code.  Honestly I should've done it much sooner, but I guess I'm just not used to that in javascript.

Reply:
totally agree